src/pages/iou/request/step/IOURequestStepDistance.tsx

```diff
@@ -94,13 +94,12 @@ function IOURequestStepDistance({
     const isLoadingRoute = transaction?.comment?.isLoading ?? false;
     const isLoading = transaction?.isLoading ?? false;
     const hasRouteError = !!transaction?.errorFields?.route;
-    const hasRoute = TransactionUtils.hasRoute(transaction, true);
+    const hasRoute = TransactionUtils.hasRoute(transaction);
     const validatedWaypoints = TransactionUtils.getValidWaypoints(waypoints);
     const previousValidatedWaypoints = usePrevious(validatedWaypoints);
     const haveValidatedWaypointsChanged = !isEqual(previousValidatedWaypoints, validatedWaypoints);
     const isRouteAbsentWithoutErrors = !hasRoute && !hasRouteError;
-    const isEmptyCoordinates = !transaction?.routes?.route0?.geometry?.coordinates?.length;
-    const shouldFetchRoute = (isEmptyCoordinates || isRouteAbsentWithoutErrors || haveValidatedWaypointsChanged) && !isLoadingRoute && Object.keys(validatedWaypoints).length > 1;
```

Comment:
@thesahindia @neil-marcellini While working on this other [PR](https://github.com/Expensify/App/pull/45575), I found an issue from our PR https://github.com/Expensify/App/pull/45139. When the selected waypoints are invalid (gives an error), it will result in an infinite request of GetRoute.

https://github.com/user-attachments/assets/a72887f9-4bf0-4c8c-baa0-0804ff4c510b


So, my first thought is to only re-fetch the route if there is no error. Then, I saw that we already have that kind of logic, the `isRouteAbsentWithoutErrors`.

`hasRoute` also checks for the route coordinate, but it also checks for `customUnit.quantity` if `isDistanceRequestType`.
https://github.com/Expensify/App/blob/52b8fddcda6177b4bf4aa9a803853a1b2670daef/src/libs/TransactionUtils.ts#L576-L578

Because `hasRoute` is only used to re-fetch the routes, I decided to remove `isEmptyCoordinates` and pass nothing to `isDistanceRequestType` param.